﻿<p>
    Test method names should clearly indicate what they are testing.
</p>
<p>
    A test’s purpose (what is being tested, any pre-conditions, and the expected result) should be obvious from its name alone.
    It should not be necessary to read the code to work out what is being tested.
</p>

<p>The general naming convention for a test class and its fixtures is:</p>

<pre>
public class &lt;ClassNameUnderTest&gt;Tests 
{
    // separated by underscores
    [Test]
    public void &lt;SystemUnderTest&gt;_&lt;PostCondition&gt;()
    {...}

    // or
    [Test]
    public void &lt;SystemUnderTest&gt;_&lt;PreCondition&gt;_&lt;PostCondition&gt;()
    {...}
}
</pre>

<p>For unit tests, the <code>SystemUnderTest</code> will usually be the name of the method you are testing.</p>

<h2>Noncompliant Code Example</h2>
<pre>
[Test]
public void HazardLightsTest()
{...}
</pre>

<h2>Compliant Code Example</h2>
<pre>
[Test]
public void HazardLightsToggleButton_WhenPushedWithLightsBlinking_StopsLightsBlinking()
{...}
</pre>

<p>
    If you are having difficultly naming your test this way, it might indicate it is doing
    too much and should be split into more granular tests.
</p>

