 <p>
     If `public` class/interface method/property return value implements `IEnumerable`, then it can _only_ be declared as one of the following:
     - `IEnumerable<T>`
     - `ISet<T>`
     - `IList<T>`
     - `IDictionary<K, V>`
     - `IReadOnlyDictionary<K, V>`
     - `KeyedCollection<K, V>`
     - `byte[]` (special case for raw binary data)
     - `string` (which happens to implement `IEnumerable<char>`)
 </p>


<h2>Noncompliant Code Example</h2>
<pre>
    public List<string> GetStrings() { ... }
    public List<string> GetStringsV2 { get; set; }
</pre>

 <h2>Compliant Code Example</h2>
<pre>
    public IList<string> GetStrings() { ... }
    public IList<string> GetStringsV2 { get; set; }
</pre>
