 <p>
     If <code>public</code> class/interface method/property return value implements <code>IEnumerable</code>, then it can <i>only</i> be declared as one of the following:
     - <code>IEnumerable&lt;T&gt;</code>
     - <code>ISet&lt;T&gt;</code>
     - <code>IList&lt;T&gt;</code>
     - <code>IDictionary&lt;K, V&gt;</code>
     - <code>IReadOnlyDictionary&lt;K, V&gt;</code>
     - <code>KeyedCollection&lt;K, V&gt;</code>
     - <code>byte[]</code> (special case for raw binary data)
     - <code>string</code> (which happens to implement `IEnumerable&lt;char&gt;`)
 </p>


<h2>Noncompliant Code Example</h2>
<pre>
    public List&lt;string&gt; GetStrings() { ... }
    public List&lt;string&gt; GetStringsV2 { get; set; }
</pre>

 <h2>Compliant Code Example</h2>
<pre>
    public IList&lt;string&gt; GetStrings() { ... }
    public IList&lt;string&gt; GetStringsV2 { get; set; }
</pre>
