﻿<p>
    Use Task.Run instead of Task.Factory.StartNew
</p>

<p>
    <ul>
        <li>Task.Factory.StartNew is considered dangerous and should only be used in very specific circumstances..</li>
        <li>The most common use-case is to specify TaskCreationOptions.LongRunning.In the current implementation of.NET this creates a new thread for your task, instead of running it on the threadpool.This could change between implementations / platforms, so should not be relied upon. Only do this if you have profiled your application and found it to be necessary.If you really need a new thread, consider Thread.Start.</li>
    </ul>
</p>

<h2>Noncompliant Code Example</h2>
<pre>
   Task.Factory.StartNew(MyMethod, CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
</pre>

<h2>Compliant Code Example</h2>
<pre>
    Task.Run(() => MyMethod());
</pre>
