<p>
    Do not use blocking task methods such as <code>Task.Wait()</code>, <code>Task.WaitAny()</code>
    and <code>Task.WaitAll()</code>
</p>
<p>
    By design, the <code>Task.Wait*</code> methods block the current thread until the task(s) complete.
    This defeats the purpose of writing async code. It also has the potential to cause deadlocks.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
    var task1 = DownloadFileAsync("...");
    var task2 = DownloadFileAsync("...");
    Task.WaitAll(task1, task2);
</pre>

<h2>Compliant Code Example</h2>
<pre>
    var task1 = DownloadFileAsync("...");
    var task2 = DownloadFileAsync("...");
    await Task.WhenAll(task1, task2); // does not block
</pre>
