﻿<p>
    Use await instead of Task.ContinueWith
</p>

 <p>
     <ul>
         <li>Task.ContinueWith has some surprising and non-intuitive behaviors, so must be used with care.</li>
         <li>await has none of these problems. It is also more readable.</li>
         <li>Therefore, Task.ContinueWith should not be used unless you are doing dynamic task parallelism, which is rare.</li>
     </ul>
 </p>

<h2>Noncompliant Code Example</h2>
<pre>
   await downloadTask.ContinueWith(async t => await SaveFileAsync(t.Result)); 
</pre>

<h2>Compliant Code Example</h2>
<pre>
    await SaveFileAsync(await downloadTask);
</pre>
