﻿<p>
    Prevent returning null from a method / property whose return value implements IEnumerable.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
public IEnumerable<int> GetProductIds(int brandId)
{
    var products = productService.GetProductsForBrand(brandId);
    
    if (products == null || !products.Any())
    {
        return null; 
        // Now the caller has to somehow know to deal with this special case.
        // You are asking for a NullReferenceException in prod.
    }
    
    return products.Select(p => p.Id);
}
</pre>

<h2>Compliant Code Example</h2>
<pre>
public IEnumerable<int> GetProductIds(int brandId)
{
    var products = productService.GetProductsForBrand(brandId);
    
    if (products == null)
    {
        // Just return an empty enumerable and everything should just work.
        return Enumerable.Empty<int>();
    }
    
    return products.Select(p => p.Id);
}
</pre>

<h2>Even better</h2>
<pre>
public IEnumerable<int> GetProductIds(int brandId)
{
    // Fix productService.GetProducts() to never return null itself, and we
    // can skip the null check entirely.
    return productService
        .GetProductsForBrand(brandId)
        .Select(p => p.Id);
}
</pre>