﻿<p>
    Use Task.Delay instead of Thread.Sleep
</p>

<p>
    <ul>
        <li>Thread.Sleep will block the current thread</li>
        <li>blocking threads has negative impact on our servers capacity, scalability and performance</li>
        <li>using Task.Delay will pause your execution but will free the thread to do other stuff.</li>
    </ul>
</p>

<h2>Noncompliant Code Examples</h2>
<pre>
   Thread.Sleep(500);
</pre>

<h2>Compliant Code Example</h2>
<pre>
    await Task.Delay(500);
</pre>