﻿<p>
    Prevent Test Fixture Inheritance.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
[TestFixture]
public class TestClass : BaseTest
{
        // Code
}
</pre>

<h2>Compliant Code Example</h2>
<pre>
[TestFixture]
public class TestClass
{
        // Code
}
</pre>