<p>
    Passing the whole <code>System.Web.HttpContext</code> to your method as a parameter create hard dependency on the HTTPContext and makes testing really hard ( mocking HTTPContext itself). 
    You should only pass parts of the <code>System.Web.HttpContext</code> that you actually using.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
using System.Web;

interface ISomething 
{
    void SomeMethod(HttpContext c, string sampleString); // ugly interface method
}
			
class TestClass: ISomething 
{
                    
    public void SomeMethod(HttpContext context, string sampleString) 
    {
    // this method is ugly
    }

    public TestClass(System.Web.HttpContext context) 
    {
        // this constructor is uglier
    }
}
</pre>

<h2>Compliant Code Example</h2>
<pre>
using System.Web;

interface ISomething 
{
    void SomeMethod(HttpContext c, string sampleString); // ugly interface method
}
			
class TestClass: ISomething 
{
                    
    public void SomeMethod(string userAgent, string sampleString) 
    {
    
    }

    public TestClass(string userAgent) 
    {
       
    }
}
</pre>
