<p>
    Long methods should be considered for refactoring
</p>
<ul>
    <li>
        This rule is to help people identify potentially long methods, so they can consider refactoring them.
    </li>
    <li>
        The rule does not count lines of whitespace in the recommended maximum is 40 lines, this varies per context though and should not be looked at as a hard limit.
    </li>
    <li>
        Try to break up long methods into smaller private methods that describe what's going on.
    </li>
    <li>
        However, don't chain methods together to avoid long methods when refactoring.
    </li>
        <h2>Noncompliant Code Example</h2>
        <pre>
class MyClass()
{
    public void MethodThatDoesALotOfStuff()
    {
        return DoThisFirst();
    }

    private int DoThisFirst()
    {
    // do some stuff...
        return DoThisSecond()
    }
    private int DoThisSecond()
    {
    // do some stuff...
        return DoThisThird();
    }
    private int DoThisThird()
    {
    // do some stuff...
    }

}
    </pre>
        <h2>Compliant Code Example</h2>
        <pre>
class MyClass()
{
    public void MethodThatDoesALotOfStuff()
    {
        DoThisFirst();
        DoThisSecond();
        return DoThisThird();
    }

    private void DoThisFirst()
    {
    // do some stuff...
    }
    private void DoThisSecond()
    {
    // do some stuff...
    }
    private int DoThisThird()
    {
    // do some stuff...
    }
}
    </pre>
        <li>
            If you are having problems with breaking methods down, try Martin Fowler's Book called refactoring https://www.amazon.com/Refactoring-Improving-Existing-Addison-Wesley-Signature/dp/0134757599/
        </li>
</ul>