<p>
    Use <code>await Task.WhenAny/All</code> instead of <code>Task.WaitAny/All</code>
</p>
<p>
    To wait for tasks to finish, use <code>await Task.WhenAll/Any</code> instead of <code>Task.WaitAll/Any</code>. 
    The latter blocks the thread while the tasks complete, defeating the purpose of using async.
</p>

<h2>Noncompliant Code Example</h2>
<code>
    var task1 = DownloadFileAsync("...");
    var task2 = DownloadFileAsync("...");
    Task.WaitAll(task1, task2);

</code>

<h2>Compliant Code Example</h2>
<code>
    var task1 = DownloadFileAsync("...");
    var task2 = DownloadFileAsync("...");
    await Task.WhenAll(task1, task2); // does not block
</code>
 